tasks/watch.cr

```diff
@@ -41,18 +43,31 @@ module Sentry
 
       self.successful_compilations += 1
       if reload_browser?
-        reload_or_start_browser_sync
+        begin
+          reload_or_start_browser_sync
+        rescue Sentry::BrowserSyncPortInUseError
+          puts "Another process is currently using the BrowserSync port #{BROWSERSYNC_PORT}".colorize(:red)
+        end
       end
     end
 
     private def reload_or_start_browser_sync
       if successful_compilations == 1
+        check_browsersync_port_availability
         start_browsersync
       else
         reload_browsersync
       end
     end
 
+    private def check_browsersync_port_availability
+      if File.executable?(`which lsof`.chomp)
+        io = IO::Memory.new
+        Process.run("lsof -i :#{BROWSERSYNC_PORT}", output: io, error: STDERR, shell: true)
```

Comment:
Can this run without `shell` or is it required?

Reply:
it's required as far as I can tell. I think I would need to know the location of the exec without the shell. Does this add a performance hit?